functests/dpdk/dpdk.go

```diff
@@ -12,69 +13,155 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
 	"k8s.io/utils/pointer"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	perfv1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	sriovk8sv1 "github.com/openshift/sriov-network-operator/pkg/apis/k8s/v1"
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	sriovtestclient "github.com/openshift/sriov-network-operator/test/util/client"
+	sriovcluster "github.com/openshift/sriov-network-operator/test/util/cluster"
+	sriovnamespaces "github.com/openshift/sriov-network-operator/test/util/namespaces"
+	sriovnetwork "github.com/openshift/sriov-network-operator/test/util/network"
 
 	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/machineconfigpool"
 	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
 )
 
-// Entry to find in the pod logfile
-const logEntry = "Accumulated forward statistics for all ports"
-const pciEnvVariableName = "PCIDEVICE_OPENSHIFT_IO_DPDKNIC"
+const (
+	SRIOV_OPERATOR_NAMESPACE = "openshift-sriov-network-operator"
+	LOG_ENTRY                = "Accumulated forward statistics for all ports"
+	PCI_ENV_VARIABLE_NAME    = "PCIDEVICE_OPENSHIFT_IO_DPDKNIC"
+	DEMO_APP_NAMESPACE       = "dpdk"
+)
+
+var (
+	TestDpdkNamespace      string
+	testDpdkImage          string
+	machineConfigPoolName  string
+	performanceProfileName string
+
+	sriovclient *sriovtestclient.ClientSet
 
-var testDpdkNamespace string
+	OriginalSriovPolicies      []*sriovv1.SriovNetworkNodePolicy
+	OriginalSriovNetworks      []*sriovv1.SriovNetwork
+	OriginalPerformanceProfile *perfv1alpha1.PerformanceProfile
+)
 
 func init() {
-	testDpdkNamespace = os.Getenv("DPDK_TEST_NAMESPACE")
-	if testDpdkNamespace == "" {
-		testDpdkNamespace = "dpdk-testing"
+	TestDpdkNamespace = os.Getenv("DPDK_TEST_NAMESPACE")
+	if TestDpdkNamespace == "" {
+		TestDpdkNamespace = "dpdk-testing"
+	}
+
+	testDpdkImage = os.Getenv("DPDK_TEST_IMAGE")
```

Comment:
Can you add this to the dockerfiles we want to build & publish on quay?

Reply:
This should be done after we merge your PR right?

Reply:
Yes